test/explicit-length-check.js

```diff
@@ -0,0 +1,65 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/explicit-length-check';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'explicit-length-check',
+	message: '`length` property should be compared to a value'
+}];
+
+ruleTester.run('explicit-length-check', rule, {
+	valid: [
+		'array.foo',
+		'array.length',
+		'array.length === 0',
+		'array.length !== 0',
+		'array.length > 0',
+		'if (array.foo) {}',
+		'if (length) {}',
+		'if ([].length > 0) {}',
+		'if ("".length > 0) {}',
+		'if (array.length === 0) {}',
+		'if (array.length !== 0) {}',
+		'if (array.length !== 0 && array[0] === 1) {}'
+	],
+	invalid: [
+		{
+			code: 'if ([].length) {}',
+			errors
```

Comment:
There isn't anything like that in the core RuleTester, so we could add that in `eslint-ava-rule-tester`, but then it'd be more than just a wrapper.

We could add that there, or add our own wrapper only in this plugin. Anyway we'd do it, I'd push against doing that, as it would make it harder for newcomers to override the default error, as they would have few examples of it in the codebase.

If we'd go for it, then I'd suggest to also do as is done for the valid cases: allow for test cases to be simple strings. Something like this

```js
const defaultError = {
  // ...
};
ruleTesterWrapper.run('rule-name', rule, {
  valid: [
    'array.foo',
    'array.length',
  ],
  invalid: [
    'if ([].length) {}',
    {
      code: 'if ([].length) {} if ([].length) {}',
      errors: [defaultError, defaultError]
    }
  ]
}, {
  errors: [defaultError]
});
```